tests/test_mdarray_ctors.cpp

```diff
@@ -181,9 +181,12 @@ void test_mdarray_ctor_data_carray() {
   free_array(errors);
 }
 
+// host data can't be used in device code
```

Comment:
What host data does the test use?  I'm guessing that the `mdarray` is defaulting to `vector` as its container; should we consider changing that to `array`, or does the test really depend on using `vector`?

Reply:
Yes, I basically meant `std::vector`. I pushed a commit using `std::array` in the test case instead which works with `SYCL` but I would guess not with `Cuda` or `HIP`.